### PR TITLE
feat(grouping): Add group tree labels to grouping components

### PR DESCRIPTION
**Pull request by jjbayer**
- Every grouping component may optionally provide a label for the corresponding node in the grouping hierarchy.
- If no label is set in a component itself, the label of one of its subcomponents is used.
- If neither the component nor its subcomponents provide a label, the label is None.

In order to use these tree labels, call `Event.get_sorted_grouping_variants()` instead of `Event.get_hashes()`.

### CODE REVIEW

File: src/sentry/eventstore/models.py [hunk 1]
```diff
@@ -362,33 +362,52 @@ def get_hashes(self, force_config=None):
...
+        return self._sort_grouping_variants(variants)
+
+    @staticmethod
+    def _hashes_from_sorted_grouping_variants(variants):
+        """ Create hashes from variants and filter out None values """
+        hashes = (variant.get_hash() for variant in variants)
+        return [hash_ for hash_ in hashes if hash_ is not None]
+
+    @staticmethod
+    def _sort_grouping_variants(variants):
```
Comment (jjbayer) [inline src/sentry/eventstore/models.py:384]: @untitaker should I move this function into a different module, as it does not depend on the Event instance?

**Review by untitaker — CHANGES_REQUESTED** [review]
one change in code + I would add this to dump_variants in the testsuite so we can see how this plays out in practice. For the latter you'll have to make up some new ascii-art/syntax that looks good and is legible

File: src/sentry/eventstore/models.py (hunk 1 above)
Comment (untitaker) [reply src/sentry/eventstore/models.py:384]: I would put the sorting fn into `sentry.grouping` but not the rest

Ideally we would actually find an API that does not require eventstore.models to call more than one function from sentry.grouping.

**Review by untitaker — CHANGES_REQUESTED** [review]
* filter components by contributes when propagating label
* do not emit tree label outside fo mobile strategy
* we'll have to rewrite this once we group by something else, but we'll just have to see

**Review by untitaker — APPROVED** [review]

**Pull request merged**